### DISCUSSION TITLE
Cookie confirming

### DISCUSSION THREAD

**Discussion by cracksilver** [automadcms/discussions#43] [Q&A]
Hi there

I'll soon be ready for the new site to go live. Now I've been thinking about cookie confirmation. How is this regulated in Automad 2? Is it still like in V1 that the system only saves the absolutely necessary cookies? I found a thread from V1 here about this: https://discuss.automad.org/d/122-cookie-note-dsgvo/5

Is it enough in this case if I make a corresponding note under data protection regulations? What if I use Google Analytics? Where can I then activate a corresponding note and confirmation button?

any help is greatly appreciated.

**Comment by marcantondahmen** [comment]
Version 1 and 2 both handle cookies in the same way. To be precise, Automad itself only creates a single PHP session cookie in order to verify if the user can edit a page. For visitors, the session is empty. It is technically possible that custom templates write to the session, which is outside the scope of this discussion here.

However, Google Analytics is a totally different kind of thing. It is a third party and therefore users must be notified and asked to accept those cookies. In case you're concerned about privacy of your visitors, I would recommend using another analytics platform that is compliant to European law.

**Reply by cracksilver** [reply]
thx. This means that there is no function in Automad that allows a popup to be activated for confirmation?

So and when not google analytics what else? We only work in Switzerland, if that matters. Google analytics is the tool what my SEO partner uses.

**Reply by marcantondahmen** [reply]
No, Automad has no tool for that. It is a CMS. Whatever the template should do must be implemented by the template developer. These are separate things. Also users should be able to decline cookies and than the page has to act accordingly. It makes much more sense that a cookie banner and analytics are implemented together.

The Swiss regulations are comparable to the European ones. And there are many compliant solutions out there.
